### PR TITLE
CMake: Search for windows message compiler

### PR DESCRIPTION
**Pull request by dnakamura**
Don't just assume that its on the path

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by DanHeidinga** [comment]
Actually, looking at the appveyor build, I see it failed due to:
```
'CMAKE_MC_COMPILER-NOTFOUND' is not recognized as an internal or external command,
1207         operable program or batch file.
```

@dnakamura Is this a path setup issue?

**Comment by dnakamura** [comment]
@DanHeidinga  The issue was that when cmake was run, `mc` was not on the path, but it was at build time. Have fixed by adding a hint which points to the windows SDK dir

**Comment by charliegracie** [comment]
@genie-omr build win xlinux

File: cmake/modules/platform/os/win.cmake [hunk 1]
```diff
@@ -36,3 +36,15 @@ list(APPEND OMR_PLATFORM_DEFINITIONS
...
 )
+
+get_filename_component(kit_dir "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows Kits\\Installed Roots;KitsRoot]" REALPATH)
+if(OMR_ENV_DATA64)
+	set(kit_dir "${kit_dir}/bin/x64")
+else()
+	set(kit_dir "${kit_dir}/bin/x86")
+endif()
+# find the message compiler
+find_program(CMAKE_MC_COMPILER mc.exe HINTS "${kit_dir}")
```
Comment (charliegracie) [inline cmake/modules/platform/os/win.cmake]: Could the `kit_dir` be some call that a user could specify if they have it installed in a random location that they do not want to have on their path?

Comment (dnakamura) [reply cmake/modules/platform/os/win.cmake]: no matter where they install it, the location should still be stored in the registry (which is where we pull `kit_dir` from). If for whatever reason its not in the registry you could still specify `-DCMAKE_MC_COMPILER=xxxx` on the command line.

**Pull request merged**